virttest/test_setup/__init__.py

```diff
@@ -300,9 +300,14 @@ def __init__(self, params):
         self.hugepage_match_str = params.get("hugepage_match_str")
         if self.hugepage_cpu_flag and self.hugepage_match_str:
             self.check_hugepage_support()
-        self.hugepage_size = self.get_hugepage_size()
-        if self.expected_hugepage_size:
-            self.check_hugepage_size_as_expected()
+        if self.kernel_hp_file == "/proc/sys/vm/nr_hugepages":
+            self.hugepage_size = self.get_hugepage_size()
+            if self.expected_hugepage_size:
+                self.check_hugepage_size_as_expected()
+        else:
+            self.hugepage_size = int(re.search(r"(\d+)kB",
+                                               self.kernel_hp_file).groups()[0])
+
```

Comment:
I think we also should raise the `ValueError` such as the `get_hugepage_size()` to prompt the root cause when it can not get the huge page size from the string of the `kernel_hp_file`.

Reply:
Agreed, has already been updated.
Thanks a lot.